src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -10,42 +15,191 @@ namespace Infrastructure.Common
+                                                           {
+                                                               return false;
+                                                           }
+                                                       });
         }
 
         public static bool SPN_Available()
         {
-            return IsConditionalTestEnabled(TestProperties.SPN_Available_PropertyName);
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // the SPN is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.SPN_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
         }
 
         public static bool Kerberos_Available()
         {
-            return IsConditionalTestEnabled(TestProperties.Kerberos_Available_PropertyName);
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // Kerberos is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.Kerberos_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
+        }
+
+        public static bool Windows_Authorization_Available()
+        {
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // Windows Authorization is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.Windows_Authorization_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
         }
 
+        public static bool NTLM_Available()
+        {
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // NTLM is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.NTLM_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
+        }
+
+        public static bool WebSockets_Supported()
```

Comment:
This eventually will become a discussion about what should be a ConditionalFact vs. ActiveIssue. In my understanding, the fact that websocket does not work on *nix is bug, which is very different from windows stream security tests will not work without some special setup (but there is no product bug).

I suggest us not to do this change in this PR (ie. continue using ActiveIssue for WebSocket for now) until we conclude [this discussion](https://github.com/dotnet/wcf/pull/1026#issuecomment-211284415).